### PR TITLE
How to create a user login page using java gui - Duplicate of #2814

### LINKED ISSUE #2578: How to create a user login page using Java GUI
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph (2-3 paragraphs):
Please write the Introductory paragraph(s), that would be included in your article. We will use this writing snippet to help us assess overall quality before approval. We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.
Take your time and write the content as you would intend to get it published. 

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Danmasanii**
I created a how-to create a user login page using java GUI index.md file inside articles folder and also added an index.md file and avatar inside authors folder.
In this article, the reader learn how to create a simple user login page in java and how to authenticate a user.
I also included a hero.png image, LoginPage.png image, and folderstructure.png image.

This closes #2578.

### CODE REVIEW

**Comment by Danmasanii** [comment]
@duncandegwa and @srishilesh, I thank you all for your assistance.
Successfully created a new pull request #2977.

**Comment by duncandegwa** [comment]
Change your author's file to `index.md` not `indexx.md`

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+### How to create a user login page using Java GUI
+
+### Introduction
+
+Java is an Object-Oriented Programming Language. G.U.I stands for Graphical User Interface. GUIs are made up of objects and classes that can interact with each other. GUIs are said to be `Event Driven` programs. this means that GUIs depends on series of object events that can occur at any given time.
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:5]: A more detailed intro could be given.

Speak more about Java. How it started as a backend framework? It's evolution into a frontend framework. What GUIs are? Where do they use it? What other languages can you build GUIs?

Give them the history behind building GUIs with Java.

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+### Prerequisites
+
+1. Basic Understanding of Java and Java GUIs.
+2. [JDK version 16 or higher](www.oracle.org).
+3. Java compiler of your choice ( Recommended: [Eclipse](www.eclipse.org) )
+
+### Setup procedure
+
+Open eclipse, click on File, select New, then select Class, name your class by Java naming convention.
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:21]: You could add a reference link detailing the above process.

Comment (srishilesh) [reply content/articles/how-to-create-a-user-login-page-using-javagui/index.md:21]: Explain the above steps better. It's hard for beginners to start with

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+If done properly you will have a window like this with the given class name, in this tutorial we name our class Java_GUI.
+
+![java gui](/how-to-create-a-user-login-page-using-java-gui/structure.png)
+**NOTE:** We will import `javax.swing.*;`,`import java.awt.event.*;` this will cover the rest of our java swing extension classes and event package.
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:25]: What is `swing` and `awt`? What does it do? Why are we using it? 

You could write separate sub topics on both

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+#### Variables
+Now let's create our private static variables class and also set it to implements an action whenever the button is click.
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:28]: This statement gives an overview. More details could be added.

Like, why private variables? What is static? What action listener are you building? Why?

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+```java
+public class Java_GUI implements ActionListener {
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:31]: `ActionListener `

Is this some inbuilt class? State why are we importing this?

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+Now let's import the JFrame extension package and create a JFrame class that will contain all the components that we will see on the window.
+
+```java
+ // JFrame class
+1 JFrame frame = new JFrame();
+2 frame.setTitle("LOGIN PAGE");
+3 frame.setLocation(new Point(500, 300));
+4 frame.add(panel);
+5 frame.setSize(new Dimension(400, 200));
+6 frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:59]: Remove these line numbers.

The readers would find it difficult to copy the codes

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+```java
+//creating a JPanel class
+JPanel panel = new JPanel();
+panel.setLayout(null);
+```
+
+#### JFrame
+
+Now let's import the JFrame extension package and create a JFrame class that will contain all the components that we will see on the window.
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:50]: What is JFrame?

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+Now let's create our private static variables class and also set it to implements an action whenever the button is click.
+
+```java
+public class Java_GUI implements ActionListener {
+
+	private static JLabel password1, label;
+	private static JTextField username;
+	private static JButton button;
+	private static JCheckBox checkbox;
+	private static JPasswordField Password;
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:37]: Explore more about JLabel, JTextField, JButton, JCheckbox, JPasswordField.

You could probably have them as sub topics.

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+#### Button
+
+```java
+// Button container
+1 button = new JButton("Login");
+2 button.setBounds(100, 110, 90, 25);
+3 button.setForeground(Color.WHITE);
+4 button.setBackground(Color.BLACK);
+5 button.addActionListener((ActionListener) new Java_GUI());
+6 panel.add(button);
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:112]: Remove line numbers

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+	JOptionPane.showMessageDialog(null, "Login Successful");
+}else {
+	JOptionPane.showMessageDialog(null, "Username or Password mismatch ");
+		}
+	}
+```
+**NOTE:** In the above block of code, we specify the username to section.io and password to 123.
+### Conclusion
+
+So far we have learned how to design a simple user login page from scratch with the help of java GUI packages. we also learned how to create an authentication method using `JOptionPane.showMessageDialog();` class.
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:136]: A more detailed summary could be given. This looks like an abrupt end.

Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:136]: Add `Further reading` section containing other reference links that the reader could refer

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+String Username = username.getText();
+String Password1 = Password.getText();
+
+if(Username.equals("section.io") && Password1.equals("123")) {
+	JOptionPane.showMessageDialog(null, "Login Successful");
+}else {
+	JOptionPane.showMessageDialog(null, "Username or Password mismatch ");
+		}
+	}
```
Comment (srishilesh) [inline content/articles/how-to-create-a-user-login-page-using-javagui/index.md:131]: Add comments.

Do this for all code snippets

**Review by srishilesh — CHANGES_REQUESTED** [review]
@Danmasanii 

Welcome to EngEd community!

Great first article. Congratulations!

Firstly, as per the guidelines, we publish articles with a min. of 750 words. Make sure you at least touch the limit.

I've suggested some major changes. Make sure to go through them.

Also, perform a Grammar check using Grammarly.

And, most importantly, from your next PR make sure not to remove the PR checklist from [here](https://github.com/section-engineering-education/engineering-education/pull/2977#issue-693041575).

Ping me when done.

Thanks

File: content/articles/how-to-create-a-user-login-page-using-javagui/index.md (hunk 11 above)
Comment (Danmasanii) [reply content/articles/how-to-create-a-user-login-page-using-javagui/index.md:131]: All noted. Thanks

**Comment by srishilesh** [comment]
@Danmasanii 

Just checking. Any updates on the suggestions?

**Comment by Danmasanii** [comment]
@srishilesh
Good day, sir, I have made the necessary changes you requested but the word count has increased to 969. what should I do now?

**Comment by srishilesh** [comment]
> @srishilesh
> Good day, sir, I have made the necessary changes you requested but the word count has increased to 969. what should I do now?

@Danmasanii 

Thanks for the changes. I'll do the review ASAP.

The minimum count of words expected is 750. So, 969 wouldn't be a problem.

**Comment by Danmasanii** [comment]
@srishilesh I will commit the changes ASAP. Thanks

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Danmasanii** [comment]
> plagiarism check done

Good day, Sir @hectorkambow, i haven't committed the changes yet. Thanks

**Comment by Danmasanii** [comment]
Good day @srishilesh, Am done. Thank you

**Review by srishilesh — APPROVED** [review]
@Danmasanii 

Thanks for making the changes.

I've made some major changes to formatting and flow. Make sure to go through my commits to understand and improve on your future articles.

Moving for the final review.

Thanks.

**Comment by srishilesh** [comment]
Duplicate of #2814

**Comment by Danmasanii** [comment]
@srishilesh thank you very much for your assistance.

**Review by louisefindlay23 — APPROVED** [review]
@Danmasanii, great article. Final review complete.

Thanks, @srishilesh for peer reviewing.

**Comment by Danmasanii** [comment]
@louisefindlay, @hectorkanbow, @srishilesh i thank you all

**Pull request merged**